File: Java/Expression.java
import java.util.Arrays;
import java.lang.Integer;
import java.util.ArrayList;

public class Expression
{
    // variáveis de instância - substitua o exemplo abaixo pelo seu próprio
    public ArrayList<String> expressions;
    public ArrayList<String[]> groups;
    public int varCount;
    
    public Expression(int [] trueValues)
    {
        ArrayList<String> expressions = new ArrayList<String>();
        this.varCount = Integer.toBinaryString(trueValues[trueValues.length - 1]).length();
        for (int i = 0; i < trueValues.length; i++){
            expressions.add(Integer.toBinaryString(trueValues[i]));
            //this.expressions[i] = Integer.toBinaryString(trueValues[i]);
        }
        this.expressions = expressions;
        int index = 0;
        ArrayList<String[]> groups = new ArrayList<String[]>();
        String [] currentGroup;
        do{
            currentGroup = groupExpressions(index);
            if (currentGroup[0] != null){
                groups.add(currentGroup);
            }
            index++;
        }while (currentGroup[0] != null);
        this.groups = groups;
    }
    
    public String[] groupExpressions(int groupIndex){
        if (expressions == null){
            return null;
        }
        String [] buffer = new String[expressions.size()];
        int bufferIndex = 0;
        for(int i = 0; i < expressions.size(); i++){
            if ((expressions.get(i).replace("0", "").length() - 1) == groupIndex){
                buffer[bufferIndex] = expressions.get(i);
                bufferIndex++;
            }
        }
        String [] result = new String[bufferIndex + 1];
        for(int i = 0; i < bufferIndex; i++){
            result[i] = buffer[i];
        }
        return result;
    }
    }

File: C#/SimplificarExpressao/SimplificarExpressao/Expression.cs
using System;

public class Expression
{
	public List<string> Expressions { get; }
	public List<List<string>> Groups { get; }
    public int VarCount { get; }
	public Expression(List<int> TrueValues) // 010 101
	{
		TrueValues.Sort();
		this.Expressions = new List<string>();
		this.VarCount = Convert.ToString(TrueValues[^1], 2).Length;
		for (int i = 0; i < TrueValues.Count; i++)
		{
			Expressions.Add(Convert.ToString(TrueValues[i], 2));
			Expressions[i] = Expressions[i].PadLeft(this.VarCount, '0');
		}
		int Index = 0;
		List<string>? CurrentGroup = new();
        this.Groups = new();
		do
		{
			CurrentGroup = GroupExpressions(Index);
			if (CurrentGroup != null )
			{
				this.Groups.Add(CurrentGroup);
			}
			Index++;
		} while (CurrentGroup != null);	

	}
	private List<string>? GroupExpressions(int GroupIndex)
	{
		string Buffer;
		List<string> GroupedExpressions = new();
		for (int i = 0; i < Expressions.Count; i++)
		{
            Buffer = Expressions[i];
			if((Buffer.Replace("0", "").Length - 1) == GroupIndex)
			{
				GroupedExpressions.Add(Expressions[i]);
			}
		}
		if (GroupedExpressions.Count == 0)
		{
			return null;
		}
		return GroupedExpressions;
	}

}
